script/utils/transferOwnership.sh

```diff
@@ -1,142 +1,193 @@
-fi
-
-echo "Using config file: $CONFIG_FILE"
-
 # Function to transfer ownership
 transfer_ownership() {
-    local contract=$1
+    local contract_name=$1
     local new_owner=$2
-    local address=$3
+    local contract_address=$3
+    local private_key=$4
+    local rpc_url=$5
+    local dry_run=$6
     
-    if [ -z "$address" ] || [ "$address" = "null" ] || [ "$address" = '""' ]; then
+    if [ -z "$contract_address" ] || [ "$contract_address" = "null" ] || [ "$contract_address" = '""' ]; then
         return
     fi
 
-    echo "Transferring ownership of $contract at $address to $new_owner"
+    echo -e "${BOLD}Transferring ownership of ${YELLOW}$contract_name${RESET}${BOLD} at $contract_address${RESET}"
     
     # Get current owner
-    current_owner=$(cast call $address "owner()" --rpc-url $TARGET_RPC_URL || echo "")
-    if [ $? -ne 0 ]; then
-        echo "Error: Failed to get current owner for $contract"
+    result=$(cast call $contract_address "owner()" --rpc-url $rpc_url)
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        echo -e "${RED}Error: Failed to get current owner for $contract_name${RESET}\nOutput: $result"
         exit 1
     fi
+    current_owner=$result
     
     if [ ! -z "$current_owner" ]; then
-        echo "Current owner: $current_owner"
+        echo -e "Current owner: ${YELLOW}$current_owner${RESET}"
         if [ "${dry_run}" = "true" ]; then
-            echo "DRY RUN: Simulating ownership transfer..."
-            if ! cast call --trace $address "transferOwnership(address)" $new_owner --rpc-url $TARGET_RPC_URL --private-key $DEPLOYER_PRIVATE_KEY; then
-                echo "Error: Dry run failed for $contract ownership transfer"
+            echo -e "${YELLOW}DRY RUN: Simulating ownership transfer...${RESET}"
+            result=$(cast call --trace $contract_address "transferOwnership(address)" $new_owner --rpc-url $rpc_url --private-key $private_key)
+            rc=$?
+            if [ $rc -ne 0 ] || [[ $result =~ "Revert" ]]; then
+                echo -e "${RED}Error: Dry run failed for $contract_name ownership transfer${RESET}\nOutput: $result"
                 exit 1
             fi
-            echo "DRY RUN: Ownership transfer simulation successful"
+            echo -e "${GREEN}DRY RUN: Ownership transfer simulation successful${RESET}"
         else
-            if ! cast send $address "transferOwnership(address)" $new_owner --rpc-url $TARGET_RPC_URL --private-key $DEPLOYER_PRIVATE_KEY; then
-                echo "Error: Failed to transfer ownership for $contract"
+            result=$(cast send $contract_address "transferOwnership(address)" $new_owner --rpc-url $rpc_url --private-key $private_key)
+            rc=$?
+            if [ $rc -ne 0 ]; then
+                echo -e "${RED}Error: Failed to transfer ownership for $contract_name${RESET}\nOutput: $result"
                 exit 1
             fi
+            echo -e "${GREEN}Successfully transferred ownership of $contract_name${RESET}"
```

Comment:
I suggest to run the call --trace always.
If it succeeds, then run the cast send if dry-run is false

Reply:
fixed in tools PR